ts/components/ui/ButtonSolid.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  StyleSheet,
+  Pressable,
+  GestureResponderEvent
+  // PixelRatio
+} from "react-native";
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withSpring,
+  useDerivedValue,
+  interpolate,
+  Extrapolate,
+  interpolateColor
+} from "react-native-reanimated";
+import themeVariables from "../../theme/variables";
+import { IOColors, IOColorType } from "../core/variables/IOColors";
+import { IOSpringValues, IOScaleValues } from "../core/variables/IOAnimations";
+import { BaseTypography } from "../core/typography/BaseTypography";
+import { IOButtonStyles } from "../core/variables/IOStyles";
+
+type Props = {
+  color?: "primary" | "danger" | "contrast";
+  label: string;
+  small?: boolean;
+  fullWidth?: boolean;
+  disabled?: boolean;
+  accessibilityLabel: string;
+  accessibilityHint?: string;
+  testID?: string;
+  onPress: (event: GestureResponderEvent) => void;
+};
+
+type ColorStates = {
+  default: string;
+  pressed: string;
+  label: {
+    default: IOColorType;
+    disabled: IOColorType;
+  };
+};
+
+// COMPONENT CONFIGURATION
+
+const mapColorStates: Record<NonNullable<Props["color"]>, ColorStates> = {
+  // Primary button
+  primary: {
+    default: IOColors.blue,
+    pressed: IOColors.blueUltraLight,
+    label: {
+      default: "white",
+      disabled: "white"
+    }
+  },
+  // Danger button
+  danger: {
+    default: IOColors.red,
+    pressed: IOColors.red,
+    label: {
+      default: "white",
+      disabled: "white"
+    }
+  },
+  // Contrast button
+  contrast: {
+    default: IOColors.white,
+    pressed: IOColors.white,
+    label: {
+      default: "blue",
+      disabled: "white"
+    }
+  }
+};
+
+// Disabled state
+const colorPrimaryButtonDisabled: IOColorType = "bluegreyLight";
+
+const styles = StyleSheet.create({
+  button: {
+    alignItems: "center",
+    textAlignVertical: "center", // Android
+    justifyContent: "center",
+    /* Legacy visual properties. They will be replaced with
+    dynamic ones once NativeBase is gone */
+    borderRadius: themeVariables.btnBorderRadius,
+    paddingHorizontal: 16,
+    // Reset default visual parameters
+    elevation: 0,
+    shadowOffset: { width: 0, height: 0 },
+    shadowRadius: 0,
+    shadowOpacity: 0
+    // Visual parameters based on the FontScale
+    // paddingVertical: PixelRatio.getFontScale() * 10,
+    // paddingHorizontal: PixelRatio.getFontScale() * 16,
+    // borderRadius: PixelRatio.getFontScale() * 8
+  },
+  backgroundDisabled: {
+    backgroundColor: IOColors[colorPrimaryButtonDisabled]
+  }
+});
+
+export const ButtonSolid = ({
+  color = "primary",
+  label,
+  small = false,
+  fullWidth = false,
+  disabled = false,
+  onPress,
+  accessibilityLabel,
+  accessibilityHint,
+  testID
+}: Props) => {
+  const isPressed: Animated.SharedValue<number> = useSharedValue(0);
+
+  // Scaling transformation applied when the button is pressed
+  const animationScaleValue = IOScaleValues?.basicButton?.pressedState;
```

Comment:
What happen if undefined?

Reply:
> What happen if undefined?

Typescript error during the build process.